api/src/components/publication/service.ts

```diff
@@ -86,14 +91,24 @@ export const updateCurrentVersion = async (id: string, updateContent: I.UpdatePu
             id: true
         }
     });
-    const updatedPublication = await client.prisma.publicationVersion.update({
+    const updatedVersion = await client.prisma.publicationVersion.update({
         where: {
             id: currentVersion?.id
         },
-        data: updateContent
+        data: dataRest
+    });
+
+    // If topics were supplied, update them on the publication (TODO: do this separately like links)
+    await client.prisma.publication.update({
```

Comment:
i think it would be better to have a separate endpoint for topics like /publications/{id}/topics where we can update/delete topics for a specific publication. Publication versions shouldn't be aware about topics

Reply:
This will be taken care of before we merge versioning into main as part of this task: https://jiscdev.atlassian.net/browse/OC-713